RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift

```diff
@@ -0,0 +1,105 @@
+import SwiftUI
+
+extension UpdateP2PLinkName.State {
+	var viewState: UpdateP2PLinkName.ViewState {
+		let (controlState, hint) = hintAndControlState
+		return .init(
+			linkName: linkName,
+			sanitizedName: sanitizedName,
+			updateButtonControlState: controlState,
+			hint: hint,
+			textFieldFocused: textFieldFocused
+		)
+	}
+
+	private var hintAndControlState: (ControlState, Hint.ViewState?) {
+		if sanitizedName != nil {
+			(.enabled, nil)
+		} else {
+			(.disabled, .iconError(L10n.LinkedConnectors.RenameConnector.errorEmpty))
+		}
+	}
+}
+
+extension UpdateP2PLinkName {
+	public struct ViewState: Equatable {
```

Comment:
Do we need `ViewState` here? Couldn't we use `UpdateP2PLinkName.State` directly?